contracts/modules/BurnV1LiquidityModule.sol

```diff
@@ -0,0 +1,46 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity 0.8.6;
+
+import "@yield-protocol/utils-v2/contracts/token/IERC20.sol";
+import "./IV1FYDai.sol";
+import "./IV1Pool.sol";
+import "../LadleStorage.sol";
+
+
+contract BurnV1LiquidityModule {
+
+    IV1Pool public immutable sepPool;
+    IV1Pool public immutable decPool;
+
+    constructor (IV1Pool sepPool_, IV1Pool decPool_) 
+    {
+        sepPool = sepPool_;
+        decPool = decPool_;
+    }
+
+    /// @dev Burns v1 tokens. If before maturity sells any fyDai for Dai, otherwise redeems fyDai for Dai
+    /// @param pool Pool to burn LP tokens from.
+    /// @param poolTokens amount of pool tokens to burn. 
+    /// @param minimumFYDaiPrice minimum Dai/fyDai price to be accepted when internally selling fyDai.
+    function migrateLiquidity(IV1Pool pool, address to, uint256 poolTokens, uint256 minimumFYDaiPrice) public {
+        require(pool == sepPool || pool == decPool, "Unknown pool");
+
+        IV1FYDai fyDai = pool.fyDai();
+
+        (uint256 daiObtained, uint256 fyDaiObtained) = pool.burn(address(this), address(this), poolTokens);
+        uint256 daiFromFYDai;
+
+        if (fyDai.maturity() > block.timestamp) {
```

Comment:
If we're only ever using two pools (sepPool and decPool) and I'm assuming `fyDai` address and the `maturity` date won't change, then would it be worth it to get these values in the constructor and store them as state variables so we don't have to call them each time?
```
uint256 sepFyDaiMaturity;
uint256 decFyDaiMaturity;


constructor( ...
   ...
   IV1FYDai sepFyDai = sepPool_.fyDai();
   sepFyDaiMatutiry = sepFyDai.maturity();   
   IV1FYDai decFyDai = sepPool_.fyDai();
   decFyDaiMatutiry = sepFyDai.maturity();
}

Reply:
The module will be delagatecalled into, so it can't have state variables (their value is in storage, and wouldn't be there when the function is delegatecalled).

We can store those as immutables, though, and they do save a call :+1: